tests/unit/s2n_server_cert_verify_test.c

```diff
@@ -14,16 +14,337 @@
+
+        struct s2n_pkey priv_key;
+        s2n_cert_type cert_type;
+
+        b.size = s2n_stuffer_data_available(&certificate_out);
+        b.data = s2n_stuffer_raw_read(&certificate_out, b.size);
+        EXPECT_SUCCESS(s2n_asn1der_to_public_key_and_type(&client_conn->secure.server_public_key, &cert_type, &b));
+
+        b.size = s2n_stuffer_data_available(&ecdsa_key_out);
+        b.data = s2n_stuffer_raw_read(&ecdsa_key_out, b.size);
+        EXPECT_SUCCESS(s2n_asn1der_to_private_key(&priv_key, &b));
+
+        EXPECT_SUCCESS(s2n_pkey_match(&client_conn->secure.server_public_key, &priv_key));
+
+        client_conn->handshake_params.our_chain_and_key->private_key = &priv_key;
+        EXPECT_NOT_NULL(client_conn->handshake_params.our_chain_and_key); 
+
+        /* Hash initialization */
+        EXPECT_SUCCESS(s2n_hash_new(&client_conn->handshake.sha256));
+        uint8_t sha256_digest_size;
+        EXPECT_SUCCESS(s2n_hash_digest_size(S2N_HASH_SHA256, &sha256_digest_size));
+        EXPECT_EQUAL(sha256_digest_size, 32);
+        EXPECT_SUCCESS(s2n_hash_init(&client_conn->handshake.sha256, S2N_HASH_SHA256));
+        EXPECT_TRUE(s2n_hash_is_ready_for_input(&client_conn->handshake.sha256));
+        EXPECT_SUCCESS(s2n_hash_update(&client_conn->handshake.sha256, hello, strlen((char *)hello)));
+        EXPECT_TRUE(s2n_hash_is_ready_for_input(&client_conn->handshake.sha256));
+        EXPECT_SUCCESS(s2n_hash_get_currently_in_hash_total(&client_conn->handshake.sha256, &bytes_in_hash));
+        EXPECT_EQUAL(bytes_in_hash, 14);
+
+        /* Send and receive cert verify */
+        EXPECT_SUCCESS(s2n_server_cert_verify_send(client_conn));
+
+        EXPECT_SUCCESS(s2n_pkey_match(&client_conn->secure.server_public_key, &priv_key));
+
+        /* Reinitialize hash */
+        EXPECT_SUCCESS(s2n_hash_init(&client_conn->handshake.sha256, S2N_HASH_SHA256));
+        EXPECT_TRUE(s2n_hash_is_ready_for_input(&client_conn->handshake.sha256));
+        EXPECT_SUCCESS(s2n_hash_update(&client_conn->handshake.sha256, hello, strlen((char *)hello)));
+        EXPECT_TRUE(s2n_hash_is_ready_for_input(&client_conn->handshake.sha256));
+        EXPECT_SUCCESS(s2n_hash_get_currently_in_hash_total(&client_conn->handshake.sha256, &bytes_in_hash));
+        EXPECT_EQUAL(bytes_in_hash, 14);
+
+        EXPECT_SUCCESS(s2n_server_cert_verify_recv(client_conn));
+
+        /* Clean up */
+        EXPECT_SUCCESS(s2n_pkey_free(&client_conn->secure.server_public_key));
+        EXPECT_SUCCESS(s2n_pkey_free(&priv_key));
+        EXPECT_SUCCESS(s2n_stuffer_free(&certificate_in));
+        EXPECT_SUCCESS(s2n_stuffer_free(&certificate_out));
+        EXPECT_SUCCESS(s2n_stuffer_free(&ecdsa_key_out));
+        EXPECT_SUCCESS(s2n_stuffer_free(&ecdsa_key_in));
+        free(cert_chain_pem);
+        free(private_key_pem);
+        EXPECT_SUCCESS(s2n_connection_free(client_conn));
+    }
+
+    /* Client errors with incorrect signed content */
```

Comment:
do you have a test that modifies the expected verify by 1 bit and test that verify fails?

Reply:
Do you think that is necessary? Incorrect signed content comes down to a difference in bits. Would you expect a different behavior from one bit off vs many?

Reply:
added anyway